misk-hibernate/src/main/kotlin/misk/hibernate/pagination/RealPager.kt

```diff
@@ -19,6 +20,8 @@ internal class RealPager<T : DbEntity<T>, Q: Query<T>>(
   }
 
   override fun nextPage(session: Session): Page<T>? {
+    @Suppress("UNCHECKED_CAST")
+    val query = query.clone<Q>()
```

Comment:
WDYT about renaming either the local val or the "original" query val?

Reply:
I did this to purposefully shadow the "original" query val, so you can't accidentally touch it in this function